File: android/src/main/java/io/adrop/bridge/AdropChannel.kt
package io.adrop.bridge

object AdropChannel {
    private const val METHOD_CHANNEL = "io.adrop.adrop-ads"
    const val invokeBannerChannel = "$METHOD_CHANNEL/banner"
    const val invokeNativeChannel = "$METHOD_CHANNEL/native"

    /**
     * Batch-loaded (preloaded) banner events — requestId-keyed, separate from
     * the tag-matched banner channel (same unitId can fill several slots).
     */
    const val invokePreloadedBannerChannel = "$METHOD_CHANNEL/preloaded_banner"

    fun invokeInterstitialChannel(id: String): String = "${METHOD_CHANNEL}/interstitial_${id}"
    fun invokeRewardedChannelOf(id: String): String = "${METHOD_CHANNEL}/rewarded_${id}"
    fun invokePopupChannelOf(id: String): String = "${METHOD_CHANNEL}/popup_${id}"
}

File: ios/bridge/AdropChannel.swift
import Foundation

struct AdropChannel {
    static let METHOD_CHANNEL = "io.adrop.adrop-ads"
    static let invokeBannerChannel = "\(METHOD_CHANNEL)/banner"
    static let invokeNativeChannel = "\(METHOD_CHANNEL)/native"
    
    static func invokeInterstitialChannel(id: String) -> String {
        return "\(METHOD_CHANNEL)/interstitial_\(id)"
    }
    
    static func invokeRewardedChannel(id: String) -> String {
        return "\(METHOD_CHANNEL)/rewarded_\(id)"
    }
    
    static func invokePopupChannel(id: String) -> String {
        return "\(METHOD_CHANNEL)/popup_\(id)"
    }
}
